Modules/Core/SpatialObjects/test/itkImageMaskSpatialObjectTest2.cxx

```diff
@@ -94,45 +94,73 @@ int itkImageMaskSpatialObjectTest2(int, char* [])
       {{ INSIDE_INDEX, INSIDE_INDEX, INSIDE_INDEX }};
     insideRegion.SetIndex( insideIndex );
     }
-
-  Iterator it( image, insideRegion );
-  it.GoToBegin();
-
-  while( !it.IsAtEnd() )
-    {
-    it.Set( itk::NumericTraits< PixelType >::max() );
-    ++it;
-    }
+  {
+    Iterator it( image, insideRegion );
+    it.GoToBegin();
+    while( !it.IsAtEnd() )
+      {
+      it.Set( itk::NumericTraits< PixelType >::max() );
+      ++it;
+      }
+  }
 
   ImageMaskSpatialObject::Pointer maskSO = ImageMaskSpatialObject::New();
-
   maskSO->SetImage(image);
   maskSO->Update();
 
+  { // Replicate use of MaskSpatialObject behavior from in itk::ImageToImageMetric.hxx
   Iterator itr( image, region );
   itr.GoToBegin();
-
   while( !itr.IsAtEnd() )
     {
-    const ImageType::IndexType constIndex =  itr.GetIndex();
-    const bool reference = insideRegion.IsInside( constIndex );
+    const ImageType::IndexType constIndex = itr.GetIndex();
 
     ImageType::PointType point;
-    image->TransformIndexToPhysicalPoint( constIndex, point );
-    const bool test      = maskSO->IsInsideInWorldSpace( point );
-    if( test != reference )
+    image->TransformIndexToPhysicalPoint(constIndex, point);
+    const bool isInsideTest = maskSO->IsInsideInWorldSpace(point);
+
+      double outsideIfZeroValue;
```

Comment:
this line and next two should have indentation level lower (indented by 2 spaces less)